### PR TITLE
Api client for /me/preferences

### PR DESCRIPTION
**Pull request by simonrozsival**
Ref #1534 

This PR adds an API client and integration tests for this endpoint.

Please run integration tests when reviewing.

Integration tests in bitrise: https://www.bitrise.io/build/8ebd928ae8712d1b

Squash whenever

### CODE REVIEW

File: Toggl.Ultrawave.Tests.Integration/PreferencesApiTests.cs [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            }
+
+            [Fact, LogTestInfo]
+            public async Task ChangesDateFormat()
+            {
+                var (togglClient, user) = await SetupTestUser();
+
+                var oldPreferences = await togglClient.Preferences.Get();
+                var newPreferences = new Preferences(oldPreferences);
+                newPreferences.DateFormat = oldPreferences.DateFormat == "YYYY-MM-DD" ? "DD.MM.YYYY" : "YYYY-MM-DD";
```
Comment (amulware) [inline Toggl.Ultrawave.Tests.Integration/PreferencesApiTests.cs]: Should we have a test for each format we think is supported to make sure they all work? Similar for other properties as applicable.

File: Toggl.Ultrawave/ApiClients/Interfaces/IPreferencesApi.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Reactive;
+using Toggl.Multivac.Models;
+
+namespace Toggl.Ultrawave.ApiClients
+{
+    public interface IPreferencesApi
+    {
+
```
Comment (amulware) [inline Toggl.Ultrawave/ApiClients/Interfaces/IPreferencesApi.cs]: 🔥

File: Toggl.Ultrawave/ApiClients/PreferencesApi.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+            : base(apiClient, serializer, credentials, endPoints.LoggedIn)
+        {
+            this.endPoints = endPoints.Preferences;
+            this.serializer = serializer;
+        }
+
+        public IObservable<IPreferences> Get()
+            => CreateObservable<Preferences>(endPoints.Get, AuthHeader);
+
+        public IObservable<Unit> Update(IPreferences preferences)
```
Comment (amulware) [inline Toggl.Ultrawave/ApiClients/PreferencesApi.cs]: Why is this an observable of unit? What does the post endpoint return?

Comment (simonrozsival) [reply Toggl.Ultrawave/ApiClients/PreferencesApi.cs]: It returns empty body, that is why this returns `Unit`.

File: Toggl.Ultrawave.Tests.Integration/PreferencesApiTests.cs (hunk 1 above)
Comment (simonrozsival) [reply Toggl.Ultrawave.Tests.Integration/PreferencesApiTests.cs]: I will add these tests.

File: Toggl.Ultrawave.Tests.Integration/PreferencesApiTests.cs [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+                newPreferences.TimeOfDayFormat = "H:m:s";
+
+                Action updatingWithWongFormat = () => togglClient.Preferences.Update(newPreferences).Wait();
+
+                updatingWithWongFormat.ShouldThrow<BadRequestException>();
+            }
+
+            [Theory, LogTestInfo]
+            [MemberData(nameof(SupportedDateFormatsTestData))]
+            public async Task SetsSpecificDateFormat(string dateFormat)
```
Comment (amulware) [inline Toggl.Ultrawave.Tests.Integration/PreferencesApiTests.cs]: 'specific'? 🔥 or explain meaning please.

File: Toggl.Ultrawave.Tests.Integration/PreferencesApiTests.cs [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+            }
+
+            [Fact, LogTestInfo]
+            public async Task ChangesDurationFormat()
+            {
+                var (togglClient, user) = await SetupTestUser();
+
+                var oldPreferences = await togglClient.Preferences.Get();
+                var newPreferences = new Preferences(oldPreferences);
+                newPreferences.DurationFormat = oldPreferences.DurationFormat == DurationFormat.Classic ? DurationFormat.Improved : DurationFormat.Classic;
```
Comment (amulware) [inline Toggl.Ultrawave.Tests.Integration/PreferencesApiTests.cs]: Let's test all three possible values please.

File: Toggl.Ultrawave/ApiClients/PreferencesApi.cs (hunk 3 above)
Comment (amulware) [reply Toggl.Ultrawave/ApiClients/PreferencesApi.cs]: I'm inquiring with backend why this is. For the purpose of this PR: 👍

**Pull request merged**